src/app/utils/field-validation/index.ts

```diff
@@ -0,0 +1,166 @@
+
+import { constructSectionValidator } from './validators/sectionValidator'
+import constructTextValidator from './validators/textValidator'
+import { ALLOWED_VALIDATORS, FIELDS_TO_REJECT } from './config'
+import fieldValidatorFactory from './FieldValidatorFactory.class'
+
+const logger = createLoggerWithLabel(module)
+
+/**
+ * Compares the response field type to the form field type
+ * @param formField The form field to compare the response to
+ * @param response The submitted response
+ */
+const isFieldTypeValid = (
+  formField: IField,
+  response: ProcessedFieldResponse,
+): Either<string, undefined> => {
+  return response.fieldType !== formField.fieldType
+    ? left(`Response fieldType (${response.fieldType}) did not match`)
+    : right(undefined)
+}
+
+/**
+ * Determines whether a response requires validation. A required field
+ * may not require an answer if it is not visible due to logic. However,
+ * if an answer is presented, it should be validated.
+ * @param formField The form field to compare the response to
+ * @param response The submitted response
+ */
+const singleAnswerRequiresValidation = (
+  formField: IField,
+  response: ProcessedSingleAnswerResponse,
+) => (formField.required && response.isVisible) || response.answer.trim() !== ''
+
+/**
+ * Generic logging function for invalid fields.
+ * Incomplete for table fields as the columnType is not logged.
+ * @param formId id of form, for logging
+ * @param formField A form field from the database
+ * @param message Message to log
+ * @throws {Error}
+ */
+const logInvalidAnswer = (
+  formId: string,
+  formField: IField,
+  message: string,
+) => {
+  logger.error({
+    message: `Invalid answer: ${message}`,
+    meta: {
+      action: 'InvalidAnswer',
+      formId,
+      fieldId: String(formField._id),
+      fieldType: formField.fieldType,
+    },
+  })
+}
+
+/**
+ * Constructs a validation function from a form field. Only meant to
+ * validate single answer responses.
+ * @param formField A form field from a form object
+ */
+const constructSingleAnswerValidator = (
+  formField: IField,
+): ResponseValidator<ProcessedSingleAnswerResponse> => {
+  if (isSectionField(formField)) {
+    return constructSectionValidator()
+  } else if (isShortTextField(formField) || isLongTextField(formField)) {
+    return constructTextValidator(formField)
+  }
+  return () => left('Unsupported field type')
+}
+
+/**
+ * Single exported function that abstracts away the complexities
+ * of field validation.
+ * @param formId id of form, for logging
+ * @param formField A form field from the database
+ * @param response A client-side response that is to be untrusted
+ * @throws
+ */
+export const validateField = (
+  formId: string,
+  formField: IField,
+  response: FieldResponse,
+): void => {
+  if (FIELDS_TO_REJECT.includes(response.fieldType)) {
+    throw new Error(`Rejected field type "${response.fieldType}"`)
```

Comment:
does this block belong in `isFieldTypeValid`?

Reply:
technically, the function signature is different. i've extracted it into its own function for now and preserving the boolean return type. i think we can refactor it to return an Either and chaining it to subsequent functions once all the field validators have been migrated.